### PR TITLE
bpo-23082: Better error message for PurePath.relative_to() from pathlib

### PR DESCRIPTION
**Pull request by rotuna**
Changed the error message to be more informative and updated the docs accordingly.

https://bugs.python.org/issue23082

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@rotuna

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Comment by rotuna** [comment]
@zooba Could you please review this PR?

File: Lib/pathlib.py [hunk 1]
```diff
@@ -922,7 +922,8 @@ def relative_to(self, *other):
         cf = self._flavour.casefold_parts
         if (root or drv) if n == 0 else cf(abs_parts[:n]) != cf(to_abs_parts):
             formatted = self._format_parsed_parts(to_drv, to_root, to_parts)
-            raise ValueError("{!r} does not start with {!r}"
+            raise ValueError("{!r} is not in the subpath of {!r}"
+                    " OR one path is relative and the other is absolute."
```
Comment (zooba) [inline Lib/pathlib.py:926]: Not convinced about the second half of the message - either it's telling me the same as the first half, or it's not telling me anything useful at all.

Perhaps we could expand the "If it's impossible" part in the docs?

**Comment by zooba** [comment]
Sorry for the delay - I had it flagged to follow-up, but life.

Pinging me on GitHub doesn't work either, btw (too many notifications; too hard to filter them). If you need my attention, please post on the bug.

File: Lib/pathlib.py (hunk 1 above)
Comment (rotuna) [reply Lib/pathlib.py:926]: The second part is not for the impossible cases. It's for cases like "/usr/home/Doc/cpython/libs" and "./Doc/cpython/libs"

I could not find a easy or clean to separate this.

**Comment by idomic** [comment]
The PR looks good pinging on the bug's page

File: Lib/pathlib.py (hunk 1 above)
Comment (zooba) [reply Lib/pathlib.py:926]: Okay, thanks for checking

**Pull request merged**